nextcord/application_command.py

```diff
@@ -745,12 +745,35 @@ def from_callback(
                 skip_counter = 1
                 # TODO: use typing.get_type_hints when 3.9 is standard
                 typehints = typing_extensions.get_type_hints(self.callback, include_extras=True)
-                # Getting the callback with `self_skip = inspect.ismethod(self.callback)` was problematic due to the
-                #  decorator going into effect before the class is instantiated, thus being a function at the time.
-                #  Try to look into fixing that in the future?
-                #  If self.parent_cog isn't reliable enough, we can possibly check if the first parameter name is `self`
-                if self.parent_cog:
-                    skip_counter += 1
```

Comment:
I think skip_counter should still be incremented if self is used, doesn't seem like this part of the code was affected.